Source/DataSources/KmlDataSource.js

```diff
@@ -226,6 +230,150 @@ define([
     var BILLBOARD_FAR_DISTANCE = 1.6093e+7;
     var BILLBOARD_FAR_RATIO = 0.1;
 
+    var kmlNamespaces = [null, undefined, 'http://www.opengis.net/kml/2.2', 'http://earth.google.com/kml/2.2', 'http://earth.google.com/kml/2.1', 'http://earth.google.com/kml/2.0'];
+    var gxNamespaces = ['http://www.google.com/kml/ext/2.2'];
+    var atomNamespaces = ['http://www.w3.org/2005/Atom'];
+    var namespaces = {
+        kml : kmlNamespaces,
+        gx : gxNamespaces,
+        atom : atomNamespaces,
+        kmlgx : kmlNamespaces.concat(gxNamespaces)
+    };
+
+    // Ensure Specs/Data/KML/unsupported.kml is kept up to date with these supported types
+    var featureTypes = {
+        Document : processDocument,
+        Folder : processFolder,
+        Placemark : processPlacemark,
+        NetworkLink : processNetworkLink,
+        GroundOverlay : processGroundOverlay,
+        PhotoOverlay : processUnsupportedFeature,
+        ScreenOverlay : processUnsupportedFeature,
+        Tour : processTour
+    };
+
+    function DeferredLoading(dataSource) {
+        this._dataSource = dataSource;
+        this._deferred = when.defer();
+        this._stack = [];
+        this._promises = [];
+        this._timeoutSet = false;
+        this._used = false;
+
+        this._started = 0;
+        this._timeThreshold = 1000; // Initial load is 1 second
+    }
+
+    defineProperties(DeferredLoading.prototype, {
+        dataSource : {
+            get : function() {
+                return this._dataSource;
+            }
+        }
+    });
+
+    DeferredLoading.prototype.addNodes = function(nodes, processingData) {
+        this._stack.push({
+            nodes: nodes,
+            index: 0,
+            processingData: processingData
+        });
+        this._used = true;
+    };
+
+    DeferredLoading.prototype.addPromise = function(promise) {
+        this._promises.push(promise);
+    };
+
+    DeferredLoading.prototype.wait = function() {
+        // Case where we had a non-document/folder as the root
+        var deferred = this._deferred;
+        if (!this._used) {
+            deferred.resolve();
+        }
+
+        return when.join(deferred.promise, when.all(this._promises));
+    };
+
+    DeferredLoading.prototype.process = function() {
+        var isFirstCall = (this._stack.length === 1);
+        if (isFirstCall) {
+            this._started = KmlDataSource._getTimestamp();
+        }
+
+        return this._process(isFirstCall);
+    };
+
+    DeferredLoading.prototype._giveUpTime = function() {
+        if (this._timeoutSet) {
+            // Timeout was already set so just return
+            return;
+        }
+
+        this._timeoutSet = true;
+        this._timeThreshold = 100; // After the first load lower threshold to 0.1 seconds
```

Comment:
Somehow my last comment got eaten because GitHub thought I wanted to do a review.  Can we change this to 50, it produces a noticeably smoother experience when loading large KML and no change in load time from what I could measure.

Reply:
In the interest of closing this out, I just made the change directly.